pkg/agent/openflow/pipeline.go

```diff
@@ -689,13 +691,21 @@ func (c *client) serviceLBBypassFlow(ipProtocol binding.Protocol) binding.Flow {
 // l2ForwardCalcFlow generates the flow that matches dst MAC and loads ofPort to reg.
 func (c *client) l2ForwardCalcFlow(dstMAC net.HardwareAddr, ofPort uint32, category cookie.Category) binding.Flow {
 	l2FwdCalcTable := c.pipeline[l2ForwardingCalcTable]
+	nextTable := ingressEntryTable
+	if ofPort == config.DefaultTunOFPort || ofPort == config.HostGatewayOFPort {
+		// Skip ingress NetworkPolicy enforcement for traffic to the
+		// tunnel or gateway interface.
+		nextTable = conntrackCommitTable
+	}
```

Comment:
Since the default next table is `conntrackCommitTable`, could we write it in this way:
```
nextTable := l2FwdCalcTable.GetNext()
if ofPort != config.DefaultTunOFPort && ofPort != config.HostGatewayOFPort {
	nextTable = ingressEntryTable
}
```
So that people know we are just making an exception for pod originated packets, and other packets just go what the pipeline tells them to go.
And could we use c.enableANP to decide the ingressEntryTable to get rid of the global variables? thought it's not good for encapsulation and unit tests.

Reply:
Sure. Updated.

For ingressEntryTable, I just do not like to duplicate the logic of deciding the table. What you think?

Reply:
I feel global variables sometimes cause obscure troubles and is kind of non object-oriented when it's mutable and required by a struct's methods and we require another function to init it first. I guess since the pipeline function is so coupled with `Client` struct, it could be its method and we could have the ingressEntryTable and egressEntryTable as member variables to avoid duplicate logic.

But no strong opinion if this beyonds your PR's scope.

Reply:
Makes sense to me to make ingressEntryTable and egressEntryTable members of client. Changed the code.